yesod-core/Yesod/Core/Class/Yesod.hs

```diff
@@ -541,18 +546,24 @@ widgetToPageContent w = HandlerFor $ \hd -> do
                    $ encodeUtf8 $ renderJavascriptUrl render s
                 return $ renderLoc x
 
-    -- modernizr should be at the end of the <head> http://www.modernizr.com/docs/#installing
-    -- the asynchronous loader means your page doesn't have to wait for all the js to load
-    let (mcomplete, asyncScripts) = asyncHelper render scripts jscript jsLoc
+    let scripts'' = catMaybes $ (\case
```

Comment:
This looks like a usage of `mapMaybe`. It's probably even more elegant as a list comprehension `[s | SOAScript s <- scripts]`.